tools/block-generator/generator/server.go

```diff
@@ -130,35 +140,26 @@ func getAccountHandler(gen Generator) func(w http.ResponseWriter, r *http.Reques
 	}
 }
 
-const blockQueryPrefix = "/v2/blocks/"
-const blockQueryBlockIdx = len(blockQueryPrefix)
-const accountsQueryPrefix = "/v2/accounts/"
-const accountsQueryAccountIdx = len(accountsQueryPrefix)
-
-func parseRound(path string) (uint64, error) {
-	if !strings.HasPrefix(path, blockQueryPrefix) {
-		return 0, fmt.Errorf("not a blocks query: %s", path)
-	}
-
-	result := uint64(0)
-	pathlen := len(path)
-
-	if pathlen == blockQueryBlockIdx {
-		return 0, fmt.Errorf("no block in path")
-	}
-
-	for i := blockQueryBlockIdx; i < pathlen; i++ {
-		if path[i] < '0' || path[i] > '9' {
-			if i == blockQueryBlockIdx {
-				return 0, fmt.Errorf("no block in path")
-			}
-			break
+func getDeltasHandler(gen Generator) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+		rd, ok := vars["round"]
+		if !ok {
+			http.Error(w, "round missing", http.StatusBadRequest)
+			return
+		}
+		round, err := strconv.ParseUint(rd, 10, 64)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
 		}
-		result = (uint64(10) * result) + uint64(int(path[i])-'0')
+		maybeWriteError(w, gen.WriteDeltas(w, round))
 	}
-	return result, nil
 }
 
+const accountsQueryPrefix = "/v2/accounts/"
+const accountsQueryAccountIdx = len(accountsQueryPrefix)
+
```

Comment:
Tech-Debt Nit:

Is there an issue outstanding for _gorilla_izing the rest of this `server.go`? If not, I recommend adding such an issue with tasks:

## Leverage Gorilla for the Block Generator Endpoints

- [ ] `/v2/accounts/{account}`
- [ ] maybe that's the only one?

Reply:
good point. let me see if there's an alternative. I just didn't want to handle the param parsing manually like this, https://github.com/shiqizng/go-algorand/blob/shiqi/block-generator/tools/block-generator/generator/server.go#L163

Reply:
maybe use [echo](https://github.com/labstack/echo)?